frontend/packages/kubevirt-plugin/src/components/modals/uninstall-kubevirt-modal/uninstall-kubevirt-message.tsx

```diff
@@ -0,0 +1,97 @@
+const NO_VMS_FOUND_MESSAGE = 'No virtual machines found on this cluster';
+
+const INCORRECT_CONFIRMATION_TEXT = 'Entered text does not match required text';
+const MISSING_CONFIRMATION_TEXT = 'Please enter confirmation text';
+
+const getNumberOfVMsOnCluster = (
+  vms: FirehoseResult<VMKind[]>,
+  vmis: FirehoseResult<VMIKind[]>,
+): number => {
+  const loadedVMs = getLoadedData(vms);
+  const loadedVMIs = getLoadedData(vmis);
+
+  const virtualMachines = _.unionBy(loadedVMs, loadedVMIs, getBasicID);
+  return virtualMachines.length;
+};
+
+const UninstallKubevirtMessageWrapped: React.FC<UninstallKubevirtMessageProps> = (props) => {
+  const { vms, vmis, setErrorMessage } = props;
+  const [confirmationTextInput, setConfirmationTextInput] = React.useState('');
+
+  const numOfVMSOnCluster = getNumberOfVMsOnCluster(vms, vmis);
+
+  const getErrorMessage = (): string => {
+    let errorMessage = '';
+    if (numOfVMSOnCluster > 1 && confirmationTextInput !== CONFIRMATION_TEXT) {
+      errorMessage =
+        confirmationTextInput === '' ? MISSING_CONFIRMATION_TEXT : INCORRECT_CONFIRMATION_TEXT;
+    }
+    return errorMessage;
+  };
+
+  const onConfirmTextInputChange = (input: string) => {
+    setConfirmationTextInput(input);
+    setErrorMessage(getErrorMessage());
+  };
+
+  const errorMessage = getErrorMessage();
+  setErrorMessage(errorMessage);
+
+  return (
+    <>
+      <p>
+        By uninstalling this operator, all virtual machines will be deleted along with their
+        storage. This action cannot be reversed and will result in data loss. Virtualization will no
+        longer be available on this cluster.
+      </p>
+      {numOfVMSOnCluster ? (
+        <>
+          <p>
+            <ExternalLink
+              href="/k8s/all-namespaces/virtualization"
+              text={`${numOfVMSOnCluster} virtual machines found on this cluster`}
+            />
+          </p>
+          <p>
+            Confirm uninstallation and deletion of all virtual machines by typing{' '}
+            <strong>{CONFIRMATION_TEXT}</strong> below:
+          </p>
+          <FormGroup
+            className="pf-m-3-col-on-md"
+            fieldId="confirm-vm-deletion-text-input"
+            id="confirm-vm-deletion-text-input"
+            isValid={!!errorMessage}
+            validated="error"
+          >
+            <TextInput
+              isRequired
+              type="text"
+              value={confirmationTextInput}
+              onChange={(v) => onConfirmTextInputChange(v)}
+              aria-label="confirmation input"
+            />
+          </FormGroup>
+        </>
+      ) : (
+        <strong>{NO_VMS_FOUND_MESSAGE}</strong>
+      )}
+    </>
+  );
+};
+
+type UninstallKubevirtMessageProps = {
+  vms?: FirehoseResult<VMKind[]>;
+  vmis?: FirehoseResult<VMIKind[]>;
+  setErrorMessage: any;
+};
+
+export const UninstallKubevirtMessage = connect(null)(UninstallKubevirtMessageWrapped);
```

Comment:
I think I'm missing something here, what does `connect(null)` do ?

Reply:
You're not missing anything. For some reason I thought that you had to pass something there. I removed it and it works fine.